packages/gitbook/src/components/DocumentView/Table/RecordRow.tsx

```diff
@@ -1,42 +1,53 @@
 import { DocumentTableViewGrid } from '@gitbook/api';
+import React from 'react';
 
 import { tcls } from '@/lib/tailwind';
 
 import { RecordColumnValue } from './RecordColumnValue';
 import { TableRecordKV, TableViewProps } from './Table';
+import { getColumnWidth } from './ViewGrid';
 
-export async function RecordRow(
+export function RecordRow(
     props: TableViewProps<DocumentTableViewGrid> & {
         record: TableRecordKV;
+        autoSizedColumns: string[];
+        fixedColumns: string[];
     },
 ) {
-    const { view } = props;
-    const columnsLengthThreshold = view.columns.length >= 7;
-
-    const tableTR = columnsLengthThreshold
-        ? ['[&>*+*]:border-l', '[&>*]:px-4']
-        : ['[&>*+*]:border-l', '[&>*+*]:pl-4'];
+    const { view, autoSizedColumns, fixedColumns } = props;
 
     return (
-        <tr className={tcls(tableTR, 'border-dark/3', 'dark:border-light/2')}>
+        <div className={tcls('flex', 'border-dark/3', 'dark:border-light/2')} role="row">
             {view.columns.map((column) => {
+                const columnWidth = getColumnWidth({
+                    column,
+                    columnWidths: view.columnWidths,
+                    autoSizedColumns,
+                    fixedColumns,
+                });
                 return (
-                    <td
+                    <div
                         key={column}
+                        role="cell"
                         className={tcls(
```

Comment:
Tables have a huge impact on the DOM / HTML size. I'm wondering if we should not use a `table.module.css` for this with:

```
.row {
  @apply flex-1 align-middle ...
}

```

It'll make the class name much smaller

Reply:
We should do it for code block lines as well at some point

Reply:
Ah ok nice good to know that table is one place where we can improve performance on. So you mean something like this: https://github.com/GitbookIO/gitbook/pull/2622/commits/2ff10758d13303be7ddf6a9100ae9e2dafb32ce2

I didnt realize applying tw classes in a css class is more efficient/performant than a bunch of individual classNames passed to the component. Is this a common known thing perhaps with a term, so I can read up on it?